File: src/main/java/de/thatsich/intellie/common/module/block/info/ABlockInfo.java
package de.thatsich.intellie.common.module.block.info;

import net.minecraft.block.Block;
import net.minecraft.block.material.Material;
import net.minecraft.creativetab.CreativeTabs;


public abstract class ABlockInfo
{
	private static final float DEFAULT_BLOCK_HARDNESS = 3.0F;
	private final ABlockName blockName;
	private final Block.SoundType blockSound;
	private final CreativeTabs creativeTab;

	protected ABlockInfo( ABlockName blockName, Block.SoundType blockSound, CreativeTabs creativeTab )
	{
		this.blockName = blockName;
		this.blockSound = blockSound;
		this.creativeTab = creativeTab;
	}

	public ABlockName getBlockName()
	{
		return this.blockName;
	}

	public Block.SoundType getBlockSound()
	{
		return this.blockSound;
	}

	public CreativeTabs getCreativeTab()
	{
		return this.creativeTab;
	}

	public Material getMaterial()
	{
		return Material.iron;
	}

	public float getHardness()
	{
		return ABlockInfo.DEFAULT_BLOCK_HARDNESS;
	}
}

File: src/main/scala/de/thatsich/minecraft/core/module/block/ABlockInfo.scala
package de.thatsich.minecraft.core.module.block

import de.thatsich.minecraft.api.mod.module.BaseTexture
import de.thatsich.minecraft.core.module.BaseName
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.creativetab.CreativeTabs

/**
 *
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
abstract class ABlockInfo( blockName: BaseName, texture: BaseTexture, hardness: Float, mat: Material, blockSound: Block.SoundType, creativeTab: CreativeTabs )
{
	def getName: BaseName =
	{
		this.blockName
	}

	def getBlockSound: Block.SoundType =
	{
		this.blockSound
	}

	def getCreativeTab: CreativeTabs =
	{
		this.creativeTab
	}

	def getMaterial: Material =
	{
		this.mat
	}

	def getHardness: Float =
	{
		this.hardness
	}

	def getTexture: BaseTexture =
	{
		this.texture
	}
}